akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -85,6 +86,32 @@ trait FileUploadDirectives {
       case Some(tuple) ⇒ provide(tuple)
       case None        ⇒ reject(MissingFormFieldRejection(fieldName))
     }
+
+  /**
+   * Collects each body part that is a multipart file as a tuple containing metadata and a `Source`
+   * for streaming the file contents somewhere. If there is no such field the request will be rejected.
+   *
+   * @group fileupload
+   */
+  def fileMultiUpload(fieldName: String): Directive1[immutable.Seq[(FileInfo, Source[ByteString, Any])]] =
+    entity(as[Multipart.FormData]).flatMap { formData ⇒
+      extractRequestContext.flatMap { ctx ⇒
+        implicit val mat = ctx.materializer
+        implicit val ec = ctx.executionContext
+
+        val partSource: Source[(FileInfo, Source[ByteString, Any]), Any] = formData.parts
+          .filter(part ⇒ part.filename.isDefined && part.name == fieldName)
+          .map(part ⇒ (FileInfo(part.name, part.filename.get, part.entity.contentType), part.entity.dataBytes))
+
+        val partF = partSource.runWith(Sink.seq[(FileInfo, Source[ByteString, Any])])
```

Comment:
Yes, of course. We'll be glad if you keep working on it. I think buffering to disk isn't completely trivial to do but would definitely be a worthwhile addition.